shipyard_blueprints/magnite/shipyard_magnite/dataclasses/budget_item.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass, field, asdict
+from typing import List, Optional, Dict, Tuple
+from shipyard_templates import ShipyardLogger
+
+logger = ShipyardLogger.get_logger()
+
+VALID_BUDGET_METRICS = {
+    "net_cost",
+    "gross_cost",
+    "requests",
+    "revenue",
+    "impressions",
+    "completes",
+    "clicks",
+}
+VALID_BUDGET_PERIODS = {"day", "lifetime", "month", "week", "hour"}
+VALID_BUDGET_PACINGS = {"asap", "smooth", "front_loaded", "even"}
+SUPPORTED_FIELDS = (
+    "id",
+    "budget_value",
+    "budget_period",
+    "budget_pacing",
+    "budget_metric",
+    "vast_caching_adjustment",
+    "budget_item_id",
+)
+
+
+@dataclass
+class BudgetItem:
```

Comment:
Should this class have an `__init__` method with the `validate` method logic inside it?

Reply:
If it weren’t for the requirement to allow partial successful uploads I would’ve agreed with you. I could have moved the validations into a `__post_init__` but since but since we're not forcing the data to be correct the ability to have the dataclass in an invalid state and mutable seemed more inline with the ask. Also made grouping the reporting easier by having the validation step be explicit. 

for context I expect there to be around 5 dataclasses as it relates to the Campaign so simpler is best in my opinion.